pyop2/utils.py

```diff
@@ -77,13 +77,16 @@ def as_tuple(item, type=None, length=None, allow_none=False):
     if configuration["type_check"]:
         if length and not len(t) == length:
             raise ValueError("Tuple needs to be of length %d" % length)
-        if type is not None:
+        if typs is not None:
+            try:
+                typs = tuple(typs)
+            except TypeError:
+                typs = tuple((typs,))
             if allow_none:
-                valid = all((isinstance(i, type) or i is None) for i in t)
-            else:
-                valid = all(isinstance(i, type) for i in t)
+                typs += (type(None),)
+            valid = all(any(isinstance(i, typ) for typ in typs) for i in t)
```

Comment:
`isinstance` accepts an iterable of things to check. So even without this patch, `as_tuple` allows one to check for more than one type.

```python
In [1]: from pyop2.utils import as_tuple

In [2]: as_tuple([1.54, 1], int)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-072f65d2b25a> in <module>
----> 1 as_tuple([1.54, 1], int)

~/tmp/firedrake/src/PyOP2/pyop2/utils.py in as_tuple(item, type, length, allow_none)
     84                 valid = all(isinstance(i, type) for i in t)
     85             if not valid:
---> 86                 raise TypeError("Items need to be of type %s" % type)
     87     return t
     88 

TypeError: Items need to be of type <class 'int'>

In [3]: as_tuple([1.54, 1], (int, float))
Out[3]: (1.54, 1)
```

Reply:
Huh, I even [used that feature myself](https://github.com/icepack/icepack/blob/master/icepack/interpolate.py#L35), I just forgot you could do that! I'll close this PR then, the issue I was having can be fixed by changing the calls within firedrake.